File: VB/Behavioural/Command/SubtractCommand.vb
Friend Class SubtractCommand
	Implements ICommand

	Private _commandValue As Double
	Sub New(commandValue As Double)
		_commandValue = commandValue
	End Sub
	Public Function DoCommand(currentValue As Double) As Double Implements ICommand.DoCommand
		Return currentValue - _commandValue
	End Function

	Public Function UndoCommand(currentValue As Double) As Double Implements ICommand.UndoCommand
		Return currentValue + _commandValue
	End Function
End Class

File: CS/Behavioural/Command/SubtractCommand.cs
namespace Pattern.Behavioural.Command
{
	internal static partial class CommandPatternRunner
	{
		internal class SubtractCommand : ICommand
        {
			private readonly double _commandValue = 0;

			public SubtractCommand(double commandValue)
			{
				_commandValue = commandValue;
			}

			public double Do(double currentValue) => currentValue -= _commandValue;
			public double Undo(double currentValue) => currentValue += _commandValue;
		}

    }
}
